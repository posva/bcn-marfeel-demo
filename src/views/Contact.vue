<template>
  <div :class="$style.red">
    <router-link to="/search">Back to Search</router-link>
    <pre class="contact">{{ contact }}</pre>
  </div>
</template>

<script>
import { contacts } from "@/assets/db.json";

export default {
  props: ["id"],
  computed: {
    contact() {
      return contacts.find(c => c.id === Number(this.id));
    }
  }
};
</script>

<style module>
.red {
  color: crimson;
}
</style>
