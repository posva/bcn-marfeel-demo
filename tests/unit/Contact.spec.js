import { shallowMount } from "@vue/test-utils";
import Contact from "@/views/Contact.vue";

describe("Contact.vue", () => {
  it("renders props.contact when passed", () => {
    const id = 3;
    const wrapper = shallowMount(Contact, {
      propsData: { id }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
