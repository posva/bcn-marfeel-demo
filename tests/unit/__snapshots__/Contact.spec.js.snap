// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact.vue renders props.contact when passed 1`] = `
<div>
  <router-link to="/search">Back to Search</router-link> <pre class="contact">{
  "id": 3,
  "firstName": "Titos",
  "lastName": "Jamot",
  "email": "tjamot2@noaa.gov",
  "gender": "Male",
  "avatar": "https://robohash.org/consequunturnostrumeligendi.png?size=250x250&amp;set=set1",
  "registeredAt": "2013-03-26",
  "isFavorite": false
}</pre></div>
`;
